apps/server/src/shared/domain/rules/user-login-migration.rule.ts

```diff
@@ -0,0 +1,24 @@
+import { Injectable } from '@nestjs/common';
+import { AuthorizationHelper } from '@src/modules/authorization/authorization.helper';
+import { AuthorizationContext, Rule } from '@src/modules/authorization/types';
+import { UserLoginMigrationDO } from '../domainobject';
+import { User } from '../entity';
+
+@Injectable()
+export class UserLoginMigrationRule implements Rule<UserLoginMigrationDO> {
```

Comment:
Maybe something more for the arc group: We shall put the repos and entities in the domain folder but this not? @alweber-cap